### PR TITLE
[botcom] Analytics tracking

### PR DESCRIPTION
**Pull request by Taha-Hassan-Git**
Added event tracking for:

- [x] File share menu - invite and publish tabs
- [x]  Anon landing page
- [x] Account menu
- [x] sidebar
- [x] Delete file dialog
- [x] rename action in file menu

### Change type

- [ ] `bugfix`
- [ ] `improvement`
- [x] `feature`
- [ ] `api`
- [ ] `other`

### Test plan

1. Create a shape...
2.

- [ ] Unit tests
- [ ] End to end tests

### Release notes

- Fixed a bug with…

### CODE REVIEW

File: apps/dotcom/client/src/tla/utils/app-ui-events.tsx [hunk 1]
```diff
@@ -8,6 +9,8 @@ export type TLAppUiEventSource =
 	| 'file-rename-dialog'
 	| 'file-menu'
 	| 'file-share-menu'
+	| 'file-header'
+	| 'anon-landing-page'
```
Comment (Taha-Hassan-Git) [inline apps/dotcom/client/src/tla/utils/app-ui-events.tsx:13]: Not sure about the source here, do we have an established name for this?

File: apps/dotcom/client/src/tla/utils/app-ui-events.tsx [hunk 2]
```diff
@@ -34,6 +39,17 @@ export interface TLAppUiEventMap {
 		padding: TldrawAppUser['exportPadding']
 		background: TldrawAppUser['exportBackground']
 	}
+	'set-shared-link-type': { type: TldrawAppFile['sharedLinkType'] | 'no-access' }
+	'learn-more-sharing-link': null
+	'learn-more-publishing': null
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/utils/app-ui-events.tsx:44]: Might be better to have a `open-url` event with the url as the payload. Something like what we do inside `tldraw`.
https://github.com/tldraw/tldraw/blob/main/packages/tldraw/src/lib/ui/context/events.tsx#L112

File: apps/dotcom/client/src/tla/components/TlaEditor/TlaEditorTopLeftPanel.tsx [hunk 3]
```diff
@@ -94,7 +96,10 @@ export function TlaEditorTopLeftPanelSignedIn() {
 		[app, fileId]
 	)
 
-	const handleRenameAction = () => setIsRenaming(true)
+	const handleRenameAction = () => {
+		setIsRenaming(true)
+		trackEvent('rename-file', { source: 'file-menu' })
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/components/TlaEditor/TlaEditorTopLeftPanel.tsx:101]: This is not the actual rename, it's just changing the local state. We might need to move this to `handleFileNameChange` or maybe even `TlaFileNameEditor`?

File: apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx [hunk 4]
```diff
@@ -215,8 +217,12 @@ function TlaSidebarFileLink({ item }: { item: RecentFile }) {
 	const { fileSlug } = useParams()
 	const isActive = TldrawAppFileRecordType.createId(fileSlug) === fileId
 	const [isRenaming, setIsRenaming] = useState(false)
+	const trackEvent = useTldrawAppUiEvents()
 
-	const handleRenameAction = () => setIsRenaming(true)
+	const handleRenameAction = () => {
+		setIsRenaming(true)
+		trackEvent('rename-file', { source: 'sidebar' })
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx:224]: Similar thing here, we probably need to move this to `TlaRenameInline`.

File: apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx [hunk 5]
```diff
@@ -321,27 +331,33 @@ function TlaSidebarFileLinkMenu({
...
+	const trackEvent = useTldrawAppUiEvents()
 	return (
 		<button
 			className={styles.toggle}
 			data-mobile={false}
 			data-testid="tla-sidebar-toggle"
-			onClick={() => updateLocalSessionState((s) => ({ isSidebarOpen: !s.isSidebarOpen }))}
+			onClick={() => {
+				updateLocalSessionState((s) => ({ isSidebarOpen: !s.isSidebarOpen }))
+				trackEvent('sidebar-toggle', { source: 'sidebar' })
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx:342]: Should we also send the new value in the data field?

File: apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx [hunk 6]
```diff
@@ -321,27 +331,33 @@ function TlaSidebarFileLinkMenu({
...
 	return (
 		<button
 			className={styles.toggle}
 			data-mobile={true}
 			data-testid="tla-sidebar-toggle-mobile"
-			onClick={() =>
+			onClick={() => {
 				updateLocalSessionState((s) => ({ isSidebarOpenMobile: !s.isSidebarOpenMobile }))
-			}
+				trackEvent('sidebar-toggle', { source: 'sidebar' })
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx:359]: Same question as above.

File: apps/dotcom/client/src/tla/layouts/TlaAnonLayout/TlaAnonLayout.tsx [hunk 7]
```diff
@@ -18,7 +20,12 @@ export function TlaAnonLayout({ children }: { children: ReactNode }) {
 				</Link>
 				<SignedOut>
 					<SignInButton mode="modal" forceRedirectUrl="/q" signUpForceRedirectUrl="/q">
-						<TlaButton data-testid="tla-signin-button">{raw('Sign in')}</TlaButton>
+						<TlaButton
+							onClick={() => trackEvent('sign-in-button', { source: 'anon-landing-page' })}
```
Comment (MitjaBezensek) [inline apps/dotcom/client/src/tla/layouts/TlaAnonLayout/TlaAnonLayout.tsx:24]: Maybe `sign-in-clicked`?

File: apps/dotcom/client/src/tla/utils/app-ui-events.tsx (hunk 1 above)
Comment (MitjaBezensek) [reply apps/dotcom/client/src/tla/utils/app-ui-events.tsx:13]: Feels ok 🤷‍♂️ Another option might be `signed-out-page`?

File: apps/dotcom/client/src/tla/components/TlaEditor/TlaEditorTopLeftPanel.tsx (hunk 3 above)
Comment (Taha-Hassan-Git) [reply apps/dotcom/client/src/tla/components/TlaEditor/TlaEditorTopLeftPanel.tsx:103]: I thought about putting it in handleFileNameChange, but assumed we wouldn't want to track every keystroke. I'll put it in handleEditingComplete within TlaFileNameEditor.

File: apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx (hunk 4 above)
Comment (Taha-Hassan-Git) [reply apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx:227]: Putting it in the handleSave function in the TlaRenameInline component, so it doesn't fire every keystroke.

**Review by MitjaBezensek — APPROVED** [review]
👏

**Pull request merged**